src/mjolnir/osmdata.cc

```diff
@@ -525,6 +562,29 @@ bool read_linguistic(const std::string& filename, LinguisticMultiMap& ling_map)
   return true;
 }
 
+bool read_conditional_speed_limits(const std::string& filename,
+                                   ConditionalSpeedLimitsMultiMap& speed_map) {
+  // Open file and truncate
+  std::ifstream file(filename, std::ios::in | std::ios::binary);
+  if (!file.is_open()) {
+    LOG_ERROR("read_conditional_speed_limits failed to open input file: " + filename);
+    return false;
+  }
+
+  // Read the count and then the temporary access restriction list
+  uint32_t count = 0;
+  file.read(reinterpret_cast<char*>(&count), sizeof(uint32_t));
+  std::vector<TempConditionalSpeedLimit> flat(count);
+  file.read(reinterpret_cast<char*>(flat.data()), count * sizeof(TempConditionalSpeedLimit));
+  file.close();
+
+  // Iterate through the temporary access restriction list and add to the restriction multi-map
+  for (const auto& l : flat) {
+    speed_map.emplace(l.way_id, l.conditional_speed_limit);
+  }
+  return true;
```

Comment:
```suggestion
  // Read the count and fill out our multimap
  uint32_t count = 0;
  file.read(reinterpret_cast<char*>(&count), sizeof(uint32_t));
  while(count--) {
    std::pair<uint64_t, ConditionalSpeedLimit> sp;
    file.read(reinterpret_cast<char*>(&sp), sizeof(sp));
    speed_map.emplace(std::move(sp));
  }  
  file.close();
  return true;
```

Reply:
its not super important but its just a bit less copying and less code to go directly from file to the multimap. you certainly dont have to refactor all the other methods that dont do this. its just a suggestion to avoid some copying and avoid the temporary structure. i guess one option would be to just use the pair isntead of the temporary structure and keep the copies to the vector. or dont do either hahaha

Reply:
Added in b08327ce2b9c8cbbbf0ac0bd8826a39f096baa17